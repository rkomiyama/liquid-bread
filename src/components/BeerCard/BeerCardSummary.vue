<template>
  <div>
    <v-card-title>
      <h1>
        {{ beer.name }}
        <span class="tagline font-italic">{{ beer.tagline }}</span>
      </h1>
    </v-card-title>
    <v-card-text class="text-xs-left pt-0">
      <v-layout>
        <v-flex xs6>
          <BeerCardValues :beer="beer"/>
        </v-flex>
        <v-flex xs6>
          <div class="font-weight-bold">Description</div>
          <div>{{ beer.description }}</div>
          <div class="font-weight-bold">Brewers tips</div>
          <div>{{ beer.brewers_tips }}</div>
          <div class="font-weight-bold">Food pairing</div>
          <div v-for="(food, i) in beer.food_pairing" :key="i">{{ food }}</div>
        </v-flex>
      </v-layout>
      <v-slide-y-transition>
        <div v-show="showMore">
          <v-divider></v-divider>
          <BeerCardDetails :ingredients="beer.ingredients" :method="beer.method"/>
        </div>
      </v-slide-y-transition>
      <v-flex text-xs-center>
        <v-btn flat @click="showMore = !showMore">{{ showMore ? 'Show Less' : 'Show More' }}</v-btn>
      </v-flex>
    </v-card-text>
  </div>
</template>

<script>
import BeerCardValues from "./BeerCardValues";
import BeerCardDetails from "./BeerCardDetails";

/**
 * Displays the description, brewers tips, and food pairing for the beer.
 */
export default {
  name: "BeerCardSummary",
  data() {
    return {
      showMore: false
    };
  },
  props: {
    /**
     * Beer data object for the beer
     */
    beer: {
      type: Object
    }
  },
  components: {
    BeerCardValues,
    BeerCardDetails
  }
};
</script>

<style lang="scss" scoped>
.tagline {
  color: grey;
}
</style>

<docs>
```jsx
const mockData = require('../../../tests/mockData/beerData.json');
<BeerCardSummary :beer="mockData" />
```
</docs>