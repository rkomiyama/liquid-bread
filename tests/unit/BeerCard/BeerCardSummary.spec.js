import Vue from "vue";
import Vuetify from "vuetify";
import { shallowMount } from "@vue/test-utils";
import { createRenderer } from "vue-server-renderer";
import BeerCardSummary from "@/components/BeerCard/BeerCardSummary.vue";
import beerData from "#/mockData/beerData.json";

Vue.use(Vuetify);

let wrapper;

describe("BeerCardSummary.vue", () => {
  beforeEach(() => {
    wrapper = shallowMount(BeerCardSummary, {
      propsData: { beer: beerData }
    });
  });

  it("BeerCardSummary has same HTML structure", () => {
    const renderer = createRenderer();
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err);
      expect(str).toMatchSnapshot();
    });
  });
});
