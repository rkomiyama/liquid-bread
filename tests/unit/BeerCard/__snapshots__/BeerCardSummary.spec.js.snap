// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerCardSummary.vue BeerCardSummary has same HTML structure 1`] = `
<div>
  <v-card-title-stub>
    <h1>
      Buzz
      <span class="tagline font-italic">A Real Bitter Experience.</span></h1>
  </v-card-title-stub>
  <v-card-text-stub class="text-xs-left pt-0">
    <v-flex-stub tag="div" xs12="" pb-2="">
      <v-card-stub tag="div" class="beer-card-summary">
        <v-layout-stub tag="div">
          <v-flex-stub tag="div" xs6="" pa-4="">
            <BeerCardValues-stub beer="[object Object]"></BeerCardValues-stub>
          </v-flex-stub>
          <v-divider-stub vertical="true"></v-divider-stub>
          <v-flex-stub tag="div" xs6="" pa-4="">
            <v-flex-stub tag="div" pb-2="">
              <div class="font-weight-bold">Description</div>
              <div>A light, crisp and bitter IPA brewed with English and American hops. A small batch brewed only once.</div>
            </v-flex-stub>
            <v-flex-stub tag="div" pb-2="">
              <div class="font-weight-bold">Brewers tips</div>
              <div>The earthy and floral aromas from the hops can be overpowering. Drop a little Cascade in at the end of the boil to lift the profile with a bit of citrus.</div>
            </v-flex-stub>
            <div class="font-weight-bold">Food pairing</div>
            <div>Spicy chicken tikka masala</div>
            <div>Grilled chicken quesadilla</div>
            <div>Caramel toffee cake</div>
          </v-flex-stub>
        </v-layout-stub>
      </v-card-stub>
    </v-flex-stub>
    <v-slide-y-transition-stub origin="top center 0">
      <BeerCardDetails-stub method="[object Object]" ingredients="[object Object]" style="display:none;"></BeerCardDetails-stub>
    </v-slide-y-transition-stub>
    <v-flex-stub tag="div" text-xs-center="">
      <v-btn-stub activeClass="v-btn--active" tag="button" flat="true" type="button">Show More</v-btn-stub>
    </v-flex-stub>
  </v-card-text-stub>
</div>
`;
